app/controllers/session_answers_controller.rb

```diff
@@ -0,0 +1,68 @@
+class SessionAnswersController < ApplicationController
+  append_view_path Rails.root.join("smart_answers")
+
+  def show
+    @title = presenter.title
+    render "smart_answers/#{page_type}", formats: [:html]
+  end
+
+  def update
+    add_new_entries_to_session
+    if presenter.current_state.error
+      render "smart_answers/#{page_type}", formats: [:html]
+    else
+      redirect_to session_flow_path(id: params[:id], node_name: next_node_name)
+    end
+  end
+
+private
+
+  def presenter
+    @presenter ||= FlowPresenter.new(session, smart_answer)
+  end
+
+  def name
+    @name ||= params[:id].gsub(/_/, "-").to_sym
+  end
+
+  def smart_answer
+    @smart_answer ||= flow_registry.find(name.to_s)
+  end
+
+  def flow_registry
+    @flow_registry ||= SmartAnswer::FlowRegistry.instance
+  end
+
+  def add_new_entries_to_session
```

Comment:
Nit: Not sure what an "entry" is - is that a response?

Reply:
Yes - response would be a better name. Thank you.